core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -637,10 +663,21 @@ public boolean hasVariableDescriptor(String variableName) {
         return effectiveVariableDescriptorMap.containsKey(variableName);
     }
 
-    public VariableDescriptor<Solution_> getVariableDescriptor(String variableName) {
+    public @Nullable VariableDescriptor<Solution_> getVariableDescriptor(String variableName) {
         return effectiveVariableDescriptorMap.get(variableName);
     }
 
+    public @NonNull VariableDescriptor<Solution_> getVariableDescriptorOrFail(String variableName) {
+        var variableDescriptor = effectiveVariableDescriptorMap.get(variableName);
+        if (variableDescriptor == null) {
+            throw new IllegalArgumentException(
+                    "Entity class %s does not hava a \"%s\" genuine or shadow variable. Maybe you meant one of %s?"
```

Comment:
Using raw strings (`"""`) will let you remove the escaping.
Also, the maybe should be on a new line, which raw strings also make easier.